lib/gooddata/lcm/actions/associate_clients.rb

```diff
@@ -46,13 +46,21 @@ def call(params)
           domain_name = params.organization || params.domain
           domain = client.domain(domain_name) || fail("Invalid domain name specified - #{domain_name}")
 
+          segments = {}
+
           params.clients.group_by { |data| data[:segment] }.each do |segment_name, clients|
             segment = domain.segments(segment_name)
+            segments[segment_name] = segment
             (clients.map(&:id) - segment.clients.map(&:id)).each do |c|
               segment.create_client(id: c)
             end
           end
 
+          params.clients.map! { |data|
```

Comment:
Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)